### PR TITLE
Crash on EAI_AGAIN (DNS lookup failure) errors

### LINKED ISSUE #857: Do not consider DNS lookup errors as FetchDocumentError
A large number of declarations in France failed [because of `EAI_AGAIN`](https://github.com/OpenTermsArchive/france-declarations/issues?q=is%3Aissue+EAI_AGAIN) errors. These seem to be linked to [DNS lookup](https://stackoverflow.com/questions/40182121/whats-the-cause-of-the-error-getaddrinfo-eai-again) failing. These should not be considered as document fetch failure but like network errors instead, crashing the app and generating an email to the administrators.

### PR DESCRIPTION
**Pull request by martinratinaud**
Fixes #857

### CODE REVIEW

File: src/archivist/fetcher/fullDomFetcher.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
-import puppeteer from 'puppeteer-extra';
+import puppeteer from 'puppeteer';
```
Comment (Ndpnt) [inline src/archivist/fetcher/fullDomFetcher.js:1]: Why did you import `puppeteer`?

From the doc of `puppeteer-extra`, it seems not necessary: 
```
// puppeteer-extra is a drop-in replacement for puppeteer,
// it augments the installed puppeteer with plugin functionality.
```

File: src/archivist/index.js [hunk 2]
```diff
@@ -130,7 +130,15 @@ export default class Archivist extends events.EventEmitter {
       }));
     } catch (error) {
       if (error instanceof FetchDocumentError) {
-        throw new InaccessibleContentError(error.message);
+        if (error.message.includes('EAI_AGAIN')) {
+          // DNS error is a network connectivity error or proxy related error. It does
+          // not necesarily mean the document is not accesible to all users but
+          // only to the Open Terms Archive engine at this specific time.
+          // As this is transient most of the time, consider it as a system error
```
Comment (Ndpnt) [inline src/archivist/index.js:137]: The problem with the sentence `It does not necesarily mean the document is not accesible to all users but only to the Open Terms Archive engine at this specific time.` is that it's true for some others errors (like 403, timed out, …) So I think we cannot use this as an argument to manage it this way.
I suggest :

```suggestion
          // EAI_AGAIN is a DNS lookup timed out error, which means it is a network connectivity error or proxy related error.
          // This operational error is mostly transient and should be handled by retrying the operation.
          // As there is no retry mechanism in the engine yet, crash the engine and leave it to the process manager to handle the retries and the delay between them.
```

**Review by Ndpnt — CHANGES_REQUESTED** [review]

File: src/archivist/fetcher/fullDomFetcher.js (hunk 1 above)
Comment (martinratinaud) [reply src/archivist/fetcher/fullDomFetcher.js:1]: puppeteer-extra does not export `errors` and especially the TimeoutError.
Could not find another way to do so

File: src/archivist/index.js (hunk 2 above)
Comment (martinratinaud) [reply src/archivist/index.js]: I agree your version is clearer, thanks

File: src/archivist/fetcher/fullDomFetcher.js (hunk 1 above)
Comment (Ndpnt) [reply src/archivist/fetcher/fullDomFetcher.js:1]: ok

**Review by Ndpnt — APPROVED** [review]

**Review by MattiSG — APPROVED** [review]
Really good comment! 👏

**Comment by martinratinaud** [comment]
Review after rebase on `expose_features`

**Review by Ndpnt — APPROVED** [review]

**Comment by Ndpnt** [comment]
Fixes #857

**Pull request merged**